eth/polls/eth_subscribe.go

```diff
@@ -26,60 +27,92 @@ type EthPoll interface {
 
 type EthSubscriptions struct {
 	polls map[string]EthPoll
+	lastPoll        map[string]uint64
+	timestamps      map[uint64][]string
 
-	lastPoll    map[string]uint64
-	timestamps  map[uint64][]string
-	lastPrune   uint64
-	evmAuxStore *evmaux.EvmAuxStore
-	blockStore  store.BlockStore
+	pollsMutex      *sync.RWMutex
+	lastPollMutex   *sync.RWMutex
+	timestampsMutex *sync.RWMutex
```

Comment:
I don't think we need one mutex per map, one to protect them all should be sufficient. The more mutexes you juggle the higher the chance of ending up with a deadlock.